### ISSUE TITLE
killing child processes

### ISSUE THREAD

**Issue by nsheff** [pepkit/geofetch#17]
If you `ctrl+c` signal a running process that has already started a `prefetch`, `geofetch` will quit but the subprocess appears to complete.

perhaps we need it to monitor the child process and kill it, too.

**Comment by nsheff** [comment]
just bit me again, so this is still an issue.

**Comment by nleroy917** [comment]
I see two places where `prefetch` is called:

https://github.com/pepkit/geofetch/blob/6f8b32a80c52c9ec957d7f609193367113b77c15/geofetch/geofetch.py#L1166
https://github.com/pepkit/geofetch/blob/6f8b32a80c52c9ec957d7f609193367113b77c15/geofetch/utils.py#L93

Does it make sense to switch all the `subprocess.call` functions to:

```python
p = subprocess.Popen(["prefetch", run_name, "--max-size", "50000000"])
PROCESS_LIST.append(p)
PROCESS_LIST[-1].wait()
...
```
since [`subprocess.call()` is just `Popen().wait()`](https://github.com/python/cpython/blob/b01cfb8f69e0990be0615b8613a7412f88bd217a/Lib/subprocess.py#L552-L566). `PROCESS_LIST` can be a global, maybe?

Then in the global [`KeyboardInterupt`](https://github.com/pepkit/geofetch/blob/50e2d272cc4a2d37756193b2334841ee076fbe0b/geofetch/geofetch.py#L2105) handler we can loop through `PROCESS_LIST` and kill all processes?

```python
for p in PROCESS_LIST:
    try:
        p.kill()
    except:
        _LOGGER.warn("Error killing process...")
```

**Comment by nleroy917** [comment]
@nsheff Is it necessary to call `.wait()` on the `psutil.Popen` call? Does `prefetch` need to run prior to executing the next lines of code?

**Comment by nleroy917** [comment]
**Some notes:**
Unlike `pypiper`, `geofetch` almost certainly doesn't need to keep an internal state of currently running child processes since these processes are blocking and must complete prior to the program to continue functioning. As such, it might make more sense to just wrap subprocess call's in a `KeyboardInterrupt` catch block. Or a more generic utility function that calls the subprocesses.

**Comment by nleroy917** [comment]
@nsheff @Khoroshevskyi Does it makes sense for the entire pipeline to be aborted on keyboard interrupt? I'm defining a wrapper to call subprocesses and gracefully terminate them + the pipeline when we received a `KeyboardInteruppt`:

```python
```

**Comment by nsheff** [comment]
> Does it makes sense for the entire pipeline to be aborted on keyboard interrupt?

Yes, that makes sense to me.

**Comment by nsheff** [comment]
> @nsheff Is it necessary to call `.wait()` on the `psutil.Popen` call? Does `prefetch` need to run prior to executing the next lines of code?

Yes.

**Comment by Khoroshevskyi** [comment]
fixed in v0.10.0
